File: staylio_backend/src/main/java/com/example/staylio_backend/dto/response/BookingHistoryResponse.java
package com.example.staylio_backend.dto.response;

import com.example.staylio_backend.model.enums.BookingStatus;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@Builder
public class BookingHistoryResponse {
    private Long bookingId;

    private String bookingCode;

    private Long roomId;

    private String roomName;

    private String imageUrl;

    private String hotelName;

    private String hotelBranchName;

    private LocalDate checkInDate;

    private LocalDate checkOutDate;

    private Integer adults;

    private Integer children;

    private BigDecimal finalPrice;

    private BookingStatus status;

    private LocalDateTime createdAt;
    
    private String paymentMethod;
    
    private String paymentUrl;
}

File: staylio_frontend/common/interfaces/response/BookingHistoryResponse.ts
import { BookingStatus } from "@common/enums/BookingStatus";

export interface BookingHistoryResponse {
    bookingId: number;
    bookingCode: number;
    roomId: number;
    roomName: string;
    imageUrl: string;
    hotelName: string;
    hotelBranchName: string;
    checkInDate: string;
    checkOutDate: string;
    adults: number;
    children: number;
    finalPrice: number;
    status: BookingStatus;
    createdAt: string;
    paymentMethod?: string;
    paymentUrl?: string;
}